File: pert04/src/main.php
<?php
declare(strict_types=1);

require_once __DIR__ . '/Pegawai.php';

// TODO Langkah 4: tambahkan Dosen dan PegawaiHarian setelah kelasnya dibuat.
$daftar = [
    new PegawaiTetap('198701012010', 'Ani Lestari', 6_000_000, 15),
    new PegawaiKontrak('K-2024-007', 'Budi Santoso', 5_000_000, 12),
    new Dosen('198901012007', 'M Rafif Akbar M.T', 7_000_000, 10, true),
    new PegawaiHarian('R-2007-003', 'Dyo', 100_000, 20),
];

echo '=== Daftar Gaji ===', PHP_EOL;
foreach ($daftar as $p) {
    echo '  ', $p, PHP_EOL;
}

$total = array_sum(array_map(fn (Pegawai $p): float => $p->hitungGaji(), $daftar));
printf('%s  Total beban gaji: Rp%s%s', PHP_EOL, number_format($total, 2, ',', '.'), PHP_EOL);

echo PHP_EOL, 'Periksa: Ani (pokok 6.000.000, masa kerja 15 tahun)', PHP_EOL;
echo '  tunjangan 15 x 2% = 30%, jadi gaji seharusnya Rp7.800.000,00', PHP_EOL;

echo PHP_EOL, 'Periksa: Budi (pokok 5.000.000, masa kerja 12 bulan)', PHP_EOL;
echo '  tunjangan 12 x 1% = 12%, jadi gaji seharusnya Rp5.600.000,00', PHP_EOL;

echo PHP_EOL, 'Periksa: Rafif (pokok 7.000.000, masa kerja 10 tahun, punya tunjangan fungsional)', PHP_EOL;
echo '  tunjangan 10 x 2% = 20%, jadi gaji seharusnya Rp8.400.000,00', PHP_EOL;

echo PHP_EOL, 'Periksa: Dyo (pokok 100.000, jam kerja 20)', PHP_EOL;
echo '  tunjangan 20 x 5.000 = 100.000, jadi gaji seharusnya Rp200.000,00', PHP_EOL;

File: pert04/src/Main.java
public class Main {
    public static void main(String[] args) {

        // TODO Langkah 4: tambahkan Dosen dan PegawaiHarian ke daftar ini
        //                 setelah Anda membuat kelasnya.
        Pegawai[] daftar = {
            new PegawaiTetap("198701012010", "Ani Lestari",  6_000_000, 15),
            new PegawaiKontrak("K-2024-007",  "Budi Santoso", 5_000_000, 12),
            new Dosen("198901012007", "M Rafif Akbar M.T", 8_000_000, 10, "Matematika"),
            new PegawaiHarian("R-2007-003", "Dyo", 50_000, 160)
        };

        System.out.println("=== Daftar Gaji ===");
        for (Pegawai p : daftar) {
            System.out.println("  " + p);
        }

        double total = 0;
        for (Pegawai p : daftar) total += p.hitungGaji();
        System.out.printf("%n  Total beban gaji: Rp%,.2f%n", total);

        System.out.println();
        System.out.println("Periksa: Ani (pokok 6.000.000, masa kerja 15 tahun)");
        System.out.println("  tunjangan 15 x 2% = 30%, jadi gaji seharusnya Rp7.800.000,00");
        System.out.println("  jenis: " + daftar[0].jenis());

        System.out.println();
        System.out.println("Periksa: Budi (gaji 5.000.000, masa kerja 12 bulan)");
        System.out.println("  gaji seharusnya Rp5.000.000,00");
        System.out.println("  jenis: " + daftar[1].jenis());

        System.out.println();
        System.out.println("Periksa: Rafif (pokok 8.000.000, masa kerja 10 tahun)");
        System.out.println("  tunjangan 10 x 2% = 20%, jadi gaji seharusnya Rp9.600.000,00");
        System.out.println("  jenis: " + daftar[2].jenis());

        System.out.println();
        System.out.println("Periksa: Dyo (upah 50.000, jam kerja 160)");
        System.out.println("  gaji seharusnya Rp8.000.000,00");
        System.out.println("  jenis: " + daftar[3].jenis());

        // Percobaan Langkah 1: hapus komentar baris berikut, kompilasi, catat pesannya.
        Pegawai langsung = new Pegawai("X", "Y", 1000) { public String jenis() { return "?"; } };
        System.out.println("Periksa: langsung (gaji 1.000)");
        System.out.println("  gaji seharusnya Rp1.000,00");
        System.out.println("  jenis: " + langsung.jenis());
    }
}